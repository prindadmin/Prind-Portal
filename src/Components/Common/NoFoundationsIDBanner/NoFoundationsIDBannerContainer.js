import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import NoFoundationsIDBanner from './NoFoundationsIDBanner'

const mapStatetoProps = state => {
  return {
    user: state.user,
  }
}

const mapDispatchToProps = dispatch => {
  return {}
}

export default withRouter(connect(mapStatetoProps, mapDispatchToProps)(NoFoundationsIDBanner))
