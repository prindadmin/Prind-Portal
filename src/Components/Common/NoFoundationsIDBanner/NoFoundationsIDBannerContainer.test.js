import { Provider, dispatch } from 'react-redux'
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import { HashRouter as Router } from "react-router-dom";

import * as Component from './NoFoundationsIDBannerContainer'

// https://www.robinwieruch.de/react-connected-component-test
const mockStore = configureStore([]);

var component;
const mockOnClosePopover = jest.fn()
const mockDispatch = jest.fn()

beforeEach(() => {
  const store = mockStore({
    user: {
      details: {
        foundationsID: "1234567890"
      }
    }
  });

  store.dispatch = mockDispatch

  const props = {};

  component = mount(
    <Provider store={store}>
      <Router>
        <Component.default {...props} />
      </Router>
    </Provider>
  );
});


it('Should render', () => {
  expect(component).toMatchSnapshot();
});
