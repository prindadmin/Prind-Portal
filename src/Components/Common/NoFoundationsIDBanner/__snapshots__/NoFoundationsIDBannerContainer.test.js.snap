// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(NoFoundationsIDBanner))>
        <Connect(NoFoundationsIDBanner)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <NoFoundationsIDBanner
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            user={
              Object {
                "details": Object {
                  "foundationsID": "1234567890",
                },
              }
            }
          />
        </Connect(NoFoundationsIDBanner)>
      </withRouter(Connect(NoFoundationsIDBanner))>
    </Router>
  </HashRouter>
</Provider>
`;
